### PR TITLE
clean up column names

### PR DESCRIPTION
**Pull request by KnathanM**
I introduced a bug where having a splits column wasn't accounted for in get column names. I also saw it was missing the weights column so added that. And generally tried to make the code more concise. Note that if there is no header row, it doesn't make sense to use any of the column CLI args (--smiles-columns, etc.) because there are no headers to reference.

### CODE REVIEW

File: chemprop/cli/utils/parsing.py [hunk 1]
```diff
@@ -74,10 +78,15 @@ def get_column_names(
...
     target_cols: Sequence[str] | None,
     ignore_cols: Sequence[str] | None,
+    splits_col: str | None,
+    weight_col: str | None,
     no_header_row: bool = False,
 ):
     df = pd.read_csv(path, header=None if no_header_row else "infer", index_col=False)
 
+    if no_header_row:
+        return ["SMILES"] + ["pred_" + str(i) for i in range((len(df.columns) - 1))]
```
Comment (shihchengli) [inline chemprop/cli/utils/parsing.py:88]: Can multicomponent be used with no_header_row set to True?

**Review by shihchengli — COMMENTED** [review]
Thanks for the fix.

File: chemprop/cli/utils/parsing.py (hunk 1 above)
Comment (KnathanM) [reply chemprop/cli/utils/parsing.py:88]: No

**Review by shihchengli — APPROVED** [review]

**Review by am2145 — APPROVED** [review]
Looks good to me. Thanks Nathan!

File: chemprop/cli/utils/parsing.py [hunk 2]
```diff
@@ -86,20 +95,18 @@ def get_column_names(
         input_cols.extend(rxn_cols)
 
     if len(input_cols) == 0:
-        if no_header_row:
-            input_cols = ["SMILES"]
-        else:
-            input_cols = [df.columns[0]]
+        input_cols = [df.columns[0]]
```
[the remark this answers is not in the document]
Comment (KnathanM) [reply chemprop/cli/utils/parsing.py:93]: I like this change. I tried to think about how to make this simplier, but didn't come up with something. This is great.

**Pull request merged**